BootloaderCorePkg/Stage2/Stage2.c

```diff
@@ -110,15 +116,33 @@ PreparePayload (
       ComponentEntry = NULL;
       LocateComponentEntry (ContainerSig, ComponentName, NULL, &ComponentEntry);
       if (ComponentEntry != NULL)  {
-        if (ComponentEntry->HashSize == SHA256_DIGEST_SIZE) {
-          TpmExtendHash (ComponentEntry->HashData, HASH_TYPE_SHA256);
+

```

Comment:
Would it be cleaner to create a separate method in this file to move all the hash extend logic into it?

Reply:
Yes, I would add separate method.